components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java

```diff
@@ -252,4 +252,34 @@ InputStream getFileById(String resourceType, String resourceName, String fileId)
     void deleteFileById(String resourceType, String resourceName, String fileId)
             throws ConfigurationManagementException;
 
+    /**
+     * This function is used to get a resource by the resource id.
+     *
+     * @param resourceId Id representing the resource.
+     * @throws ConfigurationManagementException Configuration management exception.
+     */
+    default Resource getTenantResourceById(String resourceId) throws ConfigurationManagementException {
+
+        return null;
+    }
+
+    /**
+     * This function is used to delete the given resource id.
+     *
+     * @param resourceId Request to delete the {@link Resource}.
+     * @throws ConfigurationManagementException Configuration management exception.
+     */
+    default void deleteResourceById(String resourceId) throws ConfigurationManagementException {
+
+    }
+
+    /**
+     * This function is used to replace a given resource along with all its file.
+     *
+     * @param resource resource object.
+     * @throws ConfigurationManagementException Configuration management exception.
+     */
+    default void replaceResourceAndFiles(Resource resource) throws ConfigurationManagementException {
```

Comment:
```suggestion
    default void replaceResourceWithFiles(Resource resource) throws ConfigurationManagementException {
```

Reply:
can't we simply use replaceResource for the method name here? from the description we can imply that its files will also be replaced.. WDYT

Reply:
@emswbandara, that is what should be implied from this. There is another '_replaceResource_' method which updates only the resource metadata and resource attributes.

Reply:
Changed the function name to `replaceResource(Resource resource)`. The arguments are different from the what is there in the existing replaceResource method